File: practica2/c++/over.cc
#include <stack>
#include "over.h"
#include <string>

using namespace std;

void over::ejecutar(stack<int32_t>& pila, int32_t& pc) const
{
	int32_t tmp1 = pila.top();
	pila.pop();
	int32_t tmp2= pila.top();
	pila.pop();
	pila.push(tmp2);
	pila.push(tmp1);
	pila.push(tmp2);
	pc++;
}

string over::nombre() const
{
	return "swap";
}

File: practica2/java/Over.java
import java.util.Stack;
/**
 * over
 */
public class Over extends Inst
    {

	@Override
    public int ejecutar(Stack<Integer> pila, int pc)
    {
        int tmp1 = pila.pop();
        int tmp2= pila.pop();
        pila.push(tmp2);
        pila.push(tmp1);
        pila.push(tmp2);
        pc++;
        return pc;
    }
    
    @Override
    public String nombre()
    {
        return "swap";
    }
    
}
